packages/psss-server/src/connections/MeditrakConnection.ts

```diff
@@ -3,10 +3,112 @@
  * Copyright (c) 2017 - 2020 Beyond Essential Systems Pty Ltd
  */
 
+import { generateId } from '@tupaia/database';
+import { convertPeriodStringToDateRange } from '@tupaia/utils';
 import { ApiConnection } from './ApiConnection';
 
 const { MEDITRAK_API_URL = 'http://localhost:8090/v2' } = process.env;
 
+const SURVEY_RESPONSE_FIELDS = <const>['entity.code', 'survey.code', 'submission_time', 'id'];
+const ANSWER_FIELDS = <const>['question.code', 'id'];
+const PSSS_SURVEY_RESPONSE_ANSWER_TYPE = 'Number'; // All survey response answers are 'Number' in PSSS
+
+type SurveyResponseObject = {
+  [key in typeof SURVEY_RESPONSE_FIELDS[number]]: string;
+};
+
+type AnswerObject = {
+  [key in typeof ANSWER_FIELDS[number]]: string;
+};
+
 export class MeditrakConnection extends ApiConnection {
   baseUrl = MEDITRAK_API_URL;
+
+  async findSurveyResponse(surveyCode: string, orgUnitCode: string, period: string) {
+    const [startDate, endDate] = convertPeriodStringToDateRange(period);
+    const results = (await this.get(`surveyResponses/`, {
+      page: '0',
+      pageSize: '1',
+      columns: `["${SURVEY_RESPONSE_FIELDS.join('","')}"]`,
```

Comment:
Oh I see why it's doubly helpful. Yeah still...probably better just to have some duplicate code than the extra layer of abstraction to think about when looking at the type. Makes sense to use the array here though of course! Is there a way to do it the other way around? Object.keys on a type?

Reply:
>Is there a way to do it the other way around? Object.keys on a type?

I wish!! I've actually been looking around for this for a while. There are custom libraries that do this (https://github.com/kimamula/ts-transformer-keys) but natively I don't think this is possible 😞

Reply:
Oh well. We repeat the keys in the filter again, which makes them somehow feel cheap to repeat in other places haha